### ISSUE TITLE
Add support for mapping subresource.

### ISSUE THREAD

**Issue by sailorlqh** [abcxyz/pmap#164]
### TL;DR

Our current approach for doing resource mapping uses [searchAllResource](https://cloud.google.com/asset-inventory/docs/reference/rest/v1/TopLevel/searchAllResources) api provided by [gcloud asset inventory(CAIS)](https://cloud.google.com/asset-inventory/docs/overview). We need to support resource granularity beyond this level e.g. cloud spanner table wise. Subresource is referred as those resources beyond the cloud resource level.

### Detailed design

The proposed way for supporting this is adding a new field called `subscope` in the [Resource proto](https://github.com/abcxyz/pmap/blob/main/protos/v1alpha1/resource_mapping.proto#L45-L52).

#### Subresource representation
The `subscope` field will be a string in the format of `hierarchy_string?qualifier_string`.

The `hierarchy string` will need to reflect the hierarchy of a subresource from larger granularity to narrower granularity with the format of `parent/foo/child/bar`. Example: `instance/ins/dataset/db1`.

The `qualifier_string` is a string that contains the additional information to further help determine the resources as key value pairs: `key1=value1&key2=value2`. Example: `ABC=XYZ&PROVIDER=GCP`.

#### Inheritance and how to query
Subresource will inherit the attributes (e.g. parent info, IAM policy) and annotations from its cloud resource level parent. 

When querying subresource, we consider a match only when resource.name and resource.subscope both match.

**Comment by sailorlqh** [comment]
This is reopened, as we have not yet update the integration tests.
